packages/salesforcedx-vscode-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -122,6 +127,75 @@ export class ApexReplayDebug extends DebugSession {
     this.sendResponse(response);
   }
 
+  public setBreakPointsRequest(
+    response: DebugProtocol.SetBreakpointsResponse,
+    args: DebugProtocol.SetBreakpointsArguments
+  ): void {
+    response.body = { breakpoints: [] };
+    if (args.source.path && args.lines) {
+      const uri = this.convertClientPathToDebugger(args.source.path);
+      for (const lineArg of args.lines) {
+        response.body.breakpoints.push({
+          verified: this.breakpointUtil.canSetLineBreakpoint(
+            uri,
+            this.convertClientLineToDebugger(lineArg)
+          ),
+          source: args.source,
+          line: lineArg
+        });
+      }
+    }
+    response.success = true;
+    this.sendResponse(response);
+  }
+
+  public customRequest(
+    command: string,
+    response: DebugProtocol.Response,
+    args: any
+  ): void {
+    response.success = true;
+    switch (command) {
+      case LINE_BREAKPOINT_INFO_REQUEST:
```

Comment:
Debugger extension will get line breakpoint from Apex extension. That info is passed from the debugger extension to the debugger adapter with this custom request.